src/NUnitFramework/framework/Interfaces/IAsyncCompletionStrategy.cs

```diff
@@ -0,0 +1,48 @@
+// ***********************************************************************
+// Copyright (c) 2014 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using NUnit.Framework.Internal.Commands;
+
+namespace NUnit.Framework.Interfaces
+{
+#if ASYNC
+    /// <summary>
+    /// Implement this interface to customize the waiting behavior
+    /// of the test adapter for <c>async</c> test methods
```

Comment:
"Test adapter" has a very specific meaning in the NUnit world. It's something outside the framework that __adapts__ the representation of tests in NUnit to some other form. The most obvious example is the VS Test Adapter.
In this context (i.e. within the framework itself) it has no particular meaning for me. Sorry not to suggest an alternative, but I don't know what you are trying to say.

Reply:
Looking down farther, I see somebody has added an `AwaitAdapter` in the past... I still don't like the term but I see that you aren't the one who introduced it. :confused:

Reply:
Yep, "adapter" there refers to `AwaitAdapter` and `AsyncToSyncAdapter`. I can change to something else if the wording is troublesome.